File: 0900-29-2342-max-sum-of-a-pair-with-equal-sum-of-digits/0900-29-2342-max-sum-of-a-pair-with-equal-sum-of-digits.cpp
class Solution {
public:
    int maximumSum(vector<int>& nums) {
        int n = nums.size(),ans = -1;
        int mp[82];
        for(auto x : nums){ 
            int tmp = x,sum = 0;
            while(tmp)sum+=tmp%10,tmp/=10; 
            if(!mp[sum])mp[sum] = x;
            else {
                ans=max(ans,mp[sum] + x);
                mp[sum] = max( mp[sum],x);
            }
        }
        return ans;
    }
};

File: 0900-29-2342-max-sum-of-a-pair-with-equal-sum-of-digits/0900-29-2342-max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def maximumSum(self, nums):
        ans = -1
        mp = {}

        for x in nums:
            digit_sum = sum(map(int, str(x)))
            if digit_sum in mp:
                ans = max(ans, mp[digit_sum] + x)
                mp[digit_sum] = max(mp[digit_sum], x)
            else:
                mp[digit_sum] = x

        return ans
